src/pages/reset-password.tsx

```diff
@@ -0,0 +1,218 @@
+  setError: (error: string) => void;
+};
+
+//TODO: Customize Reset password pages using continueURL
+// const ResetPassword = ({
+//   email,
+//   setStage,
+//   setError,
+// }: ResetPasswordStageProps) => {
+//   const { sendPasswordResetEmail, confirmPasswordReset } = useAuth();
+//   const [code, setCode] = useState("");
+//   const [password0, setPassword0] = useState("");
+//   const [password1, setPassword1] = useState("");
+//   const [loading, setLoading] = useState(false);
+
+//   return (
+//     <>
+//       <Text>
+//         Check your email and type the code sent to you in the box below to
+//         verify your email address. Didn't receive the email?{" "}
+//         <button
+//           onClick={(_) =>
+//             sendPasswordResetEmail(email).then(() => {
+//               //loading logic
+//             })
+//           }
+//         >
+//           <Text b className="hover:underline text-darkblue">
+//             Resend the verification email
+//           </Text>
+//         </button>
+//         .
+//       </Text>
+//       <div className="h-4" />
+//       <TitledInput
+//         title="Verification Code"
+//         placeholder="Verification Code"
+//         value={code}
+//         onChange={(e) => {
+//           setCode(e.target.value);
+//         }}
+//       />
+//       <div className="h-4" />
+//       <TitledInput
+//         title="Password"
+//         placeholder="********"
+//         value={password0}
+//         onChange={(e) => {
+//           setPassword0(e.target.value);
+//         }}
+//       />
+//       <TitledInput
+//         title="Re-Enter Password"
+//         placeholder="********"
+//         value={password1}
+//         onChange={(e) => {
+//           setPassword1(e.target.value);
+//         }}
+//       />
+//       <div className="h-6" />
+//       <div className="flex justify-between">
+//         <Button
+//           disabled={loading}
+//           size="small"
+//           variant="inverted"
+//           onClick={() => {
+//             setError("");
+//             setStage(ResetPasswordStages.InputEmail);
+//           }}
+//         >
+//           Back
+//         </Button>
+//         <Button
+//           disabled={loading}
+//           size="small"
+//           onClick={(_) => {
+//             if (password0 === password1) {
+//               setLoading(true);
+//               confirmPasswordReset(code, password0).then(() => {
+//                 /* Success Message & Log in */
+//                 setError("");
+//                 setStage(ResetPasswordStages.SuccessfulReset);
+//                 setLoading(false);
+//               });
+//             } else {
+//               setError("Make sure the password fields match.");
+//             }
+//           }}
+//         >
+//           Reset Password
+//         </Button>
+//       </div>
+//     </>
+//   );
+// };
+
+const InputEmail = ({
+  email,
+  setEmail,
```

Comment:
Specific callback functions shouldn't be passed into components. Your component should not touch business logic as it is simply an input field.

For example, instead of passing in `setEmail`, a better pattern would be to pass in an onChange function:

```
<Input onChange={setEmail} />
```

and then in your component,

```
<input onChange={onChange} />
```

`onChange` is not specific to business logic and helps your component just be a component.